coalib/settings/ConfigurationGathering.py

```diff
@@ -105,8 +107,12 @@ def warn_nonexistent_targets(targets, sections, log_printer):
     :param sections:    The sections to search. (Dict.)
     :param log_printer: The log printer to warn to.
     """
+
+    sections_substrings = list(itertools.chain(
+        *[sections_name.split('.') for sections_name in sections]))
```

Comment:
Could you change this to

```python
sections_substrings = {
    substring
    for sections_name in sections
    for substring in sections_name.split('.')
}
```

?

I feel like it's waaay cleaner. And `set` is a more appropriate data type here than `list`.